library/src/cache/patch_manager.rs

```diff
@@ -275,6 +275,19 @@ impl PatchManager {
             }
         }
     }
+
+    /// Deletes all patch artifacts with numbers less than patch_number.
+    fn delete_patch_artifacts_older_than(&mut self, patch_number: usize) -> Result<()> {
+        for entry in std::fs::read_dir(self.patches_dir())? {
+            let entry = entry?;
+            let number = entry.file_name().to_string_lossy().parse::<usize>()?;
+            if number < patch_number {
+                let _ = self.delete_patch_artifacts(number);
+            }
```

Comment:
Do you want this API to just delete all others? Is there ever a case where you want to keep around "newer" ones?

Reply:
We want to keep around newer patches. Consider:

1. We start booting patch 2
2. While booting (i.e., in between boot start and boot success), we download and inflate patch 3
3. We finish booting patch 2

Deleting all other patch artifacts would delete patch 3, and because we've "seen" patch 3, we would never try to download it again (it would be considered "bad").

Adding a comment.